qemu/tests/numa_prealloc_threads.py

```diff
@@ -43,27 +62,49 @@ def run(test, params, env):
     if not thread_id:
         test.fail("No thread-id setted.")
 
-    expected_cpu_affinity = thread_context_device.get_param("cpu-affinity")
+    error_context.context("Check the CPU affinity", test.log.info)
+    expected_cpu_affinity = thread_context_device.get_param("cpu-affinity", "0")
     cpu_affinity = vm.monitor.qom_get(thread_context_device_id, "cpu-affinity")
-    affinity = str(cpu_affinity[0]) + "-" + str(cpu_affinity[-1])
+    if cpu_affinity[0] != 0:
+        affinity = str(cpu_affinity[0]) + "-" + str(cpu_affinity[-1])
+    else:
+        affinity = "0"
     test.log.debug("The affinity: %s and the expected_cpu_affinity: %s"
                    % (affinity, expected_cpu_affinity))
     if expected_cpu_affinity != affinity:
         test.fail("Test and QEMU cpu-affinity does not match!")
 
-    cmd_taskset = "taskset -c -p " + str(thread_id)
-    output = process.getoutput(cmd_taskset)
-    if not re.search(affinity, output):
-        test.fail("The affinities %s and %s do not match!"
-                  % (affinity, str(output)))
+    cmd_taskset = "taskset -c -p %s" % thread_id
+    check_affinity(affinity, cmd_taskset, test)
 
     sandbox = params.get("qemu_sandbox", "on")
     if sandbox == "on":
-        try:
-            # The command is expected to fail
-            vm.monitor.qom_set(thread_context_device_id, "cpu-affinity", cpu_affinity)
-        except QMPCmdError as e:
-            test.log.debug("The expected error message: %s and the output: %s"
-                           % (error_msg, e.data))
-            if not re.search(error_msg, str(e.data)):
-                test.fail("Can not get expected error message: %s" % error_msg)
+        error_context.base_context("Setting a new affinity with sandbox enabled",
+                                   test.log.info)
+    try:
+        # The command is expected to fail when sandbox=on
+        vm.monitor.qom_set(thread_context_device_id,
+                           "cpu-affinity",
+                           first_cpu_affinity)
+        # Updates affinity with the latest value
+        affinity = str(first_cpu_affinity[0]) + "-" + str(first_cpu_affinity[-1])
+    except QMPCmdError as e:
+        error_context.context("The expected error message: %s and the output: %s"
+                              % (error_msg, e.data), test.log.debug)
+        if not re.search(error_msg, str(e.data)):
+            test.fail("Cannot get expected error message: %s" % error_msg)
+
+    if operation_type != "boot_cpu_affinity":
+        check_affinity(affinity, cmd_taskset, test)
```

Comment:
This checking is valid for case `boot_cpu_affinity` as my understanding, cam be moved out of the if blocker.

Reply:
Updated, thanks!